src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/retryfailedmanagedindex/RetryFailedManagedIndexRequest.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement.transport.action.retryfailedmanagedindex
+
+import org.elasticsearch.action.ActionRequest
+import org.elasticsearch.action.ActionRequestValidationException
+import org.elasticsearch.action.ValidateActions
+import org.elasticsearch.common.io.stream.StreamInput
+import org.elasticsearch.common.io.stream.StreamOutput
+import java.io.IOException
+import java.util.Collections
+
+class RetryFailedManagedIndexRequest : ActionRequest {
+
+    val indices: List<String>
+    val startState: String?
+
+    constructor(
+        indices: List<String>,
+        startState: String?
+    ) : super() {
+        this.indices = indices
+        this.startState = startState
+    }
+
+    @Throws(IOException::class)
+    constructor(sin: StreamInput) : this(
+        indices = Collections.unmodifiableList(sin.readStringList()),
```

Comment:
Probably asked before but I keep forgetting, why are we using Collections.umnodifiableList here?

Reply:
emmm, it seems this is not necessary, copy and paste from Sriram's actually